### PR TITLE
CLDR-17109 st: Actually remove the rest of Derby

### PR DESCRIPTION
**Pull request by srl295**
remove derbyNotSupported(), db_Derby, AND db_Mysql AND getDBKind()

CLDR-17109

- [X] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/DBUtils.java [hunk 1]
```diff
@@ -79,49 +68,35 @@ public static final void derbyNotSupported() {
...
+    public static final String DB_SQL_ENGINE_INNO = "ENGINE=InnoDB";
+    public static final String DB_SQL_MB4 = " CHARACTER SET utf8mb4 COLLATE utf8mb4_bin";
     public static int db_number_open = 0;
     public static int db_number_used = 0;
-    private static int db_UnicodeType = java.sql.Types.VARCHAR; /*
-                                                                * for setNull -
-                                                                * see
-                                                                * java.sql.Types
-                                                                */
+    private static final int db_UnicodeType = java.sql.Types.BLOB;
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/DBUtils.java:99]: On first glance this seems to go beyond dropping derby -- does this potentially change the behavior when using mysql? Now I see setupSqlForServerType had override for mysql... so I'm assuming there's no actual change for mysql

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/DBUtils.java [hunk 2]
```diff
@@ -378,38 +310,19 @@ public boolean tableExists(String table) {
...
-                                canonColumn)
-                        > 0;
-            }
-        } catch (SQLException se) {
-            SurveyMain.busted("While looking for column '" + table + "." + column + "': ", se);
-            return false; // NOTREACHED
-        }
+        return sqlCount(
+                        conn,
+                        "select count(*) from information_schema.COLUMNS where table_name=? and column_name=?",
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/DBUtils.java:322]: ideally all caps for the sql reserved words -- out of scope for this ticket though

**Review by btangmu — APPROVED** [review]
Hurray!

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/DBUtils.java (hunk 1 above)
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/DBUtils.java:99]: Yes it moves the overrides up. I'm trying to make it easier to see what the MySQL settings are.

**Pull request merged**